File: Algorithms/1342. Number of Steps to Reduce a Number to Zero/javascript/solution.counting-bits.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
    const binary = num.toString(2);
    const ones = binary.split('').filter((x) => x === '1').length;
    return ones + binary.length - 1;
};

File: Algorithms/1342. Number of Steps to Reduce a Number to Zero/golang/solution.counting-bits.go
import "strconv"

func numberOfSteps(num int) int {
	binary := strconv.FormatInt(int64(num), 2)
	var ones int = 0
	for _, v := range binary {
		if v == '1' {
			ones++
		}
	}
	return ones + len(binary) - 1
}